### PR TITLE
[GFX940] Add winograd support

### PR DESCRIPTION
**Pull request by Slimakanzer**
This PR adds Winograd support for GFX94X ASICs. The `FORCE_CACHE_BYPASS_ON_STORE` has been added to the solver's kernel build parameters to control cache bypass.

## Local verification
@muralinr tested the shader functionality on the gfx940 device, using a small number of test cases for each kernel configuration.

### CODE REVIEW

File: src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s [hunk 1]
```diff
@@ -28,7 +28,11 @@
 .if (.amdgcn.gfx_generation_number == 9)
     KERNEL_PROLOG fp16_dot2_edc_f2x3_dilation2
 
+.if (.amdgcn.gfx_generation_number == 9 && .amdgcn.gfx_generation_minor == 4 && .amdgcn.gfx_generation_stepping == 0)
```
Comment (atamazov) [inline src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s]: [Recommendation]
```suggestion
.if (.amdgcn.gfx_generation_minor == 4 && .amdgcn.gfx_generation_stepping == 0)
```
The same for other similar checks.

File: src/solver/conv_winoRxS.cpp [hunk 2]
```diff
@@ -711,6 +711,7 @@ ConvSolution ConvBinWinoRxS<Winodata, Winofilter>::GetSolution(
 
     KernelBuildParameters options{
         {"ROCM_METADATA_VERSION", 5},
+        {"FORCE_CACHE_BYPASS_ON_STORE", StartsWith(name, "gfx940")},
```
Comment (atamazov) [inline src/solver/conv_winoRxS.cpp]: Let's discuss this offline

**Comment by atamazov** [comment]
@Slimakanzer Let's enable smoke testing of these winograd solvers on gfx940 in test/CmakeLists.txt

**Review by junliume — DISMISSED** [review]

File: src/solver/conv_winoRxS.cpp [hunk 3]
```diff
@@ -709,8 +709,11 @@ ConvSolution ConvBinWinoRxS<Winodata, Winofilter>::GetSolution(
     kernel.l_wk.push_back(1);
     kernel.l_wk.push_back(1);
 
+    const auto force_cache_bypass = StartsWith(name, "gfx940") || StartsWith(name, "gfx941");
```
Comment (atamazov) [inline src/solver/conv_winoRxS.cpp]: [Recommendation] We do not need StartsWith when comparing for equality.
```suggestion
    const auto force_cache_bypass = (name == "gfx940") || (name == "gfx941");
```
[Q] So we expect both 940 and 941 require cache bypass?

File: src/solver/conv_winoRxS_fused.cpp [hunk 4]
```diff
@@ -219,8 +219,11 @@ ConvSolution ConvBinWinogradRxSf2x3g1Fused::GetSolution(const FusionContext& con
     kernel.l_wk.push_back(1);
     kernel.l_wk.push_back(1);
 
+    const auto force_cache_bypass = StartsWith(name, "gfx940") || StartsWith(name, "gfx941");
```
Comment (atamazov) [inline src/solver/conv_winoRxS_fused.cpp]: [R] Ditto

File: src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s (hunk 1 above)
Comment (atamazov) [reply src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s]: Resolved.

File: src/solver/conv_winoRxS.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_winoRxS.cpp]: Resolved.

**Review by atamazov — DISMISSED** [review]
This can be merged, **_but please add the enabled solvers to #2101_**, so we'll track [this comment](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2083#issuecomment-1502415119) there.

File: src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s (hunk 1 above)
Comment (atamazov) [reply src/kernels/Conv_Winograd_v30_2_6_fp16_dot2_f2x3_dilation2.s]: Resolved

File: src/solver/conv_winoRxS.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_winoRxS.cpp]: Resolved

File: src/solver/conv_winoRxS_fused.cpp (hunk 4 above)
Comment (Slimakanzer) [reply src/solver/conv_winoRxS_fused.cpp]: Thank you, fixed in [be7bb7f](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2083/commits/be7bb7f0673f2d789965e016942516e121e2ad3e)

**Review by atamazov — APPROVED** [review]
LGTM, please add the enabled solvers to #2101, so we'll track [this comment](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2083#issuecomment-1502415119) there.

**Comment by Slimakanzer** [comment]
@Kirpich30000 please review

**Review by muralinr — APPROVED** [review]
LGTM

**Review by junliume — APPROVED** [review]

**Pull request merged**